src/eom-util.c

```diff
@@ -427,3 +427,47 @@ eom_util_show_file_in_filemanager (GFile *file, GtkWindow *toplevel)
 	if (!done)
 		_eom_util_show_file_in_filemanager_fallback (file, toplevel);
 }
+
+gboolean
+eom_notebook_page_scroll_event_cb (GtkWidget         *widget,
+                                   GdkEventScroll    *event)
+
+{
+    GtkNotebook *notebook = GTK_NOTEBOOK (widget);
+    GtkWidget *child, *event_widget, *action_widget;
+
+    child = gtk_notebook_get_nth_page (notebook, gtk_notebook_get_current_page (notebook));
+    if (child == NULL)
+        return FALSE;
+
+    event_widget = gtk_get_event_widget ((GdkEvent*) event);
+
+    /* Ignore scroll events from the content of the page */
+    if (event_widget == NULL || event_widget == child || gtk_widget_is_ancestor (event_widget, child))
+        return FALSE;
+
+    /* And also from the action widgets */
+    action_widget = gtk_notebook_get_action_widget (notebook, GTK_PACK_START);
+    if (event_widget == action_widget || (action_widget != NULL && gtk_widget_is_ancestor (event_widget, action_widget)))
+        return FALSE;
+
+    action_widget = gtk_notebook_get_action_widget (notebook, GTK_PACK_END);
+    if (event_widget == action_widget || (action_widget != NULL && gtk_widget_is_ancestor (event_widget, action_widget)))
+        return FALSE;
+
+    switch (event->direction)
+    {
+        case GDK_SCROLL_RIGHT:
+        case GDK_SCROLL_DOWN:
+            gtk_notebook_next_page (notebook);
+            break;
+        case GDK_SCROLL_LEFT:
+        case GDK_SCROLL_UP:
+            gtk_notebook_prev_page (notebook);
+            break;
+        case GDK_SCROLL_SMOOTH:
+        break;
```

Comment:
You can also handle  GDK_SCROLL_SMOOTH.
See https://github.com/mate-desktop/mate-terminal/commit/622e2d1272bf2d7a99d147482ae6a51b0a77f188

Reply:
Yes, i have seen your other PR about this subject.
There is another PR i purpose about notebook scroll event in the mate-system-monitor.
I'll follow your code to update it.

Reply:
Done.

Reply:
Do not mark conversations as resolved.

Reply:
why didn't you use the same code to handle GDK_SCROLL_SMOOTH direction?

Reply:
code indent will be fixed on #277